kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/eq/IterableEq.kt

```diff
@@ -0,0 +1,21 @@
+package io.kotest.assertions.eq
+
+import io.kotest.assertions.Actual
+import io.kotest.assertions.Expected
+import io.kotest.assertions.failure
+import io.kotest.assertions.show.show
+
+object IterableEq : Eq<Iterable<*>> {
+   override fun equals(actual: Iterable<*>, expected: Iterable<*>): Throwable? {
+      val actualAsList = actual.toList()
```

Comment:
if performance matters, `while` statement can be used instead of `toList`.

```
while (actual.hasNext() && expected.hasNext()) {
        int comparison = actual.next().compareTo(expected.next());
        if (comparison != 0) {
            return failure(Expected(expected.show()), Actual(actual.show()))
        }
    }
return null
```

Reply:
I prefer the code base to be more FP style, and a while loop with an early return is not. This is definitely faster, but micro optimizations in a test framework are less important than more robust code imo.